### PR TITLE
Fixed Incorrect Current Cap Space Logic

### PR DESCRIPTION
**Pull request by Lyddon7522**
#17 - Fix Current Cap Space Logic calculation

Added Unit Tests
Updated Teams Detail and List API to pass date.

### CODE REVIEW

File: DynamoLeagueBlazor/Shared/Utilities/CapSpaceUtilities.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ public static class CapSpaceUtilities
     public static int GetCurrentCapValue(DateOnly date)
         => (date.Year - StartYear) / 2 * BiYearlyCapIncrease + BaseCapSpace;
 
-    public static int CalculateCurrentCapSpace(int rosteredPlayersContractValue, int unrosteredPlayersContractValue, int unsignedPlayeredContractValue)
-        => rosteredPlayersContractValue + unsignedPlayeredContractValue + (unrosteredPlayersContractValue / 2);
+    public static int CalculateCurrentCapSpace(DateOnly date, int rosteredPlayersContractValue, int unrosteredPlayersContractValue, int unsignedPlayeredContractValue)
```
[the remark this answers is not in the document]
Comment (Lyddon7522) [reply DynamoLeagueBlazor/Shared/Utilities/CapSpaceUtilities.cs:12]: I like this. I'll I would update `GetCurrentCapValue` to `GetCurrentSalaryCap`

File: DynamoLeagueBlazor/Tests/Utilities/CapSpaceTests.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    {
+        private Team? stubTeam;
+        private int rosteredPlayersContractValue;
+        private int unrosteredPlayersContractValue;
+        private int unsignedPlayersContractValue;
+
+        [SetUp]
+        public void Init()
+        {
+            stubTeam = CreateFakeTeam();
```
[the remark this answers is not in the document]
Comment (Lyddon7522) [reply DynamoLeagueBlazor/Tests/Utilities/CapSpaceTests.cs]: @benjaminsampica updated

File: DynamoLeagueBlazor/Tests/Utilities/CapSpaceTests.cs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        [Test]
+        public void GivenRosteredPlayers_WhenTheYearIs2021AndRosteredContractIs500_ThenReturnRemaingCapOf550()
+        {
+            var date = new DateOnly(2021, 01, 01);
+
+            var expectedResult = 550;
+            var result = CapSpaceUtilities.CalculateCurrentCapSpace(date, rosteredPlayersContractValue, 0, 0);
+            result.Should().Be(expectedResult);
+        }
+        [Test]
```
[the remark this answers is not in the document]
Comment (Lyddon7522) [reply DynamoLeagueBlazor/Tests/Utilities/CapSpaceTests.cs]: @benjaminsampica Updated

File: DynamoLeagueBlazor/Shared/Utilities/CapSpaceUtilities.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lyddon7522) [reply DynamoLeagueBlazor/Shared/Utilities/CapSpaceUtilities.cs]: @benjaminsampica updated

File: DynamoLeagueBlazor/Tests/Utilities/CapSpaceTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lyddon7522) [reply DynamoLeagueBlazor/Tests/Utilities/CapSpaceTests.cs]: Also, thanks for the tip. This looks so much better now.

**Review by benjaminsampica — APPROVED** [review]

**Pull request merged**